jumpscale/packages/vdc_dashboard/chats/etcd.py

```diff
@@ -69,17 +70,56 @@ def initializing(self):
             self.k8s_client.delete_deployed_release(self.release_name)
             self.stop(dedent(stop_message))
 
+        self.is_certified = False
+        start_request_time = time()
+        request = None
+        j.logger.debug("ETCD Check certificate")
+        while time() - start_request_time <= timeout:
+            try:
+                request = j.tools.http.get(f"https://{self.domain}", timeout=timeout, verify=True)
+                self.is_certified = True
+                break
+            except j.tools.http.exceptions.SSLError:
+                self.md_show_update(
+                    dedent(
+                        f"Initializing your {self.SOLUTION_TYPE}...<br />\n### Check domain certificate should take around 2 to 3 minutes... <br />\n"
+                    ),
+                    md=True,
+                )
+                gevent.sleep(10)
+
+        if not self.is_certified:
+            request = j.tools.http.get(f"https://{self.domain}", timeout=timeout, verify=False)
+
+        # Etcd using TCP, always response return this msg "404 page not found"
+        if not "404 page not found" in str(request.content):
```

Comment:
we can check for the status_code instead, to be 404 ?

Reply:
It is preferred to let it check with content, as this response will be different if it fails for anything else.